tests/test_filters.py

```diff
@@ -86,6 +92,70 @@ def test_longdate_datetime_with_tz(self):
+            assert filters.date_filter(self.date, format='medium') == u'31 जन॰ 2020'
+
+    def test_date_localized_long_hi(self):
+        with self.app.test_request_context('/', headers={'Accept-Language': 'hi'}):
+            assert filters.date_filter(self.date, format='long') == u'31 जनवरी 2020'
+
+    def test_time_localized_hi_medium(self):
+        with self.app.test_request_context('/', headers={'Accept-Language': 'hi'}):
+            assert filters.datetime_filter(self.datetime, format='medium') == u'31 जन॰ 2020, 12:00:00 am'
+
+    def test_month_localized_hi(self):
+        with self.app.test_request_context('/', headers={'Accept-Language': 'hi'}):
+            assert filters.date_filter(self.date, "MMMM") == u'जनवरी'
+
+    def test_month_localized_en(self):
+        with self.app.test_request_context('/'):
+            assert filters.date_filter(self.date, "MMMM") == u'January'
+
+    def test_time_localized_short(self):
+        with self.app.test_request_context('/', headers={'Accept-Language': 'hi'}):
+            assert filters.time_filter(self.datetime, format='short') == u'12:00 am'
+
+    def test_time_localized_medium(self):
+        with self.app.test_request_context('/', headers={'Accept-Language': 'hi'}):
+            assert filters.time_filter(self.datetime, format='medium') == u'12:00:00 am'
+
+    def test_time_localized_long(self):
+        with self.app.test_request_context('/', headers={'Accept-Language': 'hi'}):
+            assert filters.time_filter(self.datetime, format='long') == u'12:00:00 am UTC'
+
+    def test_time_localized_hi_full(self):
+        with self.app.test_request_context('/', headers={'Accept-Language': 'hi'}):
+            assert filters.time_filter(self.time, format='full') == u'11:59:59 pm समन्वित वैश्विक समय'
+
+    def test_time_localized_en_full(self):
+        with self.app.test_request_context('/', headers={'Accept-Language': 'en'}):
+            assert filters.time_filter(self.time, format='full') == u'11:59:59 PM Coordinated Universal Time'
+
+    def test_datetime_with_usertz(self):
+        with self.app.test_request_context('/'):
+            assert filters.datetime_filter(self.datetimeEST, format='full', usertz=False) == u'Friday, January 31, 2020 at 12:00:00 AM Eastern Standard Time'
+
+    def test_datetime_without_usertz(self):
+        with self.app.test_request_context('/'):
+            assert filters.datetime_filter(self.datetimeEST, format='full', usertz=True) == u'Friday, January 31, 2020 at 4:56:00 AM Coordinated Universal Time'
+
+    def test_date_dmy(self):
+        with self.app.test_request_context('/'):
+            assert filters.date_filter(self.datetime, format='short', locale='en_GB') == u'31/01/2020'
+
+    def test_date_mdy(self):
+        with self.app.test_request_context('/'):
+            assert filters.date_filter(self.datetime, format='short', locale='en_US') == u'1/31/20'
```

Comment:
The locale should be through the Accept Language header. We're clearly missing language variants here.